ui/src/hooks/useVerifyProof.ts

```diff
@@ -1,48 +1,18 @@
-import { useEffect } from 'react'
-import { GetFunctionArgs } from 'viem'
-import { useContractRead } from 'wagmi'
-import { abi, bigintify, config } from '#'
 import { useAsync, useStore } from '@/hooks'
-
-const { groth16 } = require('snarkjs')
+import { useSpartanEcdsaWorker } from './useSpartanEcdsaWorker'
 
 export const useVerifyProof = () => {
-  const { proof, publicSignals } = useStore()
-
-  const {
-    data,
-    error: errorSnarkJs,
-    execute,
-    isLoading: isLoadingSnarkJs,
-  } = useAsync<string>(
-    () => groth16.exportSolidityCallData(proof, publicSignals),
-    'verify-proof-calldata',
-  )
-  const args: GetFunctionArgs<typeof abi, 'verifyProof'>['args'] =
-    data !== undefined ? bigintify(JSON.parse(`[${data}]`)) : undefined
+  const { proof } = useStore()
+  const { verifyMembership } = useSpartanEcdsaWorker()
 
-  useEffect(() => {
-    execute()
-  }, [execute])
+  const { data: isValid, error } = useAsync(async () => {
```

Comment:
capture: #348 

What I was doing was weird / not right :thinking:  
`useAsync` wraps `useSWR` which is designed as a hook for data fetching (it includes a caching mechanism).
I think I was wrongly using it for any async stuff.
Let's define a separate hook for promises / async stuff that don't do data fetching.

Not blocking merging this PR as this is existing problem I am just realizing now

Reply:
That's right that needs refinement.